governance/API-Ref-Guidelines-Update.md

```diff
@@ -0,0 +1,508 @@
+
+As of the time of writing this guide, the customer for the API reference are white-glove customers and internal studios using O3DE. Zero of these customers have called out any specific interest in reading generated API Reference guides, and in general, they all prefer to read header files and source code to understand APIs.
+
+Note: We have configured Doxygen to ignore "docs.h" files, which solely contain documentation. Docs for specific APIs will be moved to the corresponding header files, and extensive documentation will be moved to the [O3DE Developer Guide](https://o3de.org/docs/).
+
+## How much documentation should I add?
+
+Doxygen distinguishes between two types of descriptions: brief descriptions and detailed descriptions. With our configuration, a brief description is the first sentence of a Doxygen-formatted code comment (described later). A detailed description is anything that follows the first sentence.
+
+Do not use `@brief` or `/brief`. Our doxygen configuration automatically assumes the first sentence (ending with a period) is the brief description in a comment block or multiline comment.
+
+In general, when deciding what to document and how much, just consider the customer whenever adding documentation.  The customer is primarily an SDE reading the public API headers and trying to quickly figure out what APIs are available, what they do, and how to use them. This means that information that is redundant and provides no extra information is of negative value, as it will clutter the header and make it harder to read. But information that is useful to a reader such as input constraints, explanation of a class's purpose, references to other related structures, explanation of parameters that have idiosyncrasies, and other explanations that help a reader of the header understand how to use the API are all recommended.
+
+Avoid adding _too much_ documentation to API headers about the inner workings.  You can reserve deep explorations of the code for the .cpp file itself, if necessary.
+
+## Can I exclude code from being included in the documentation?
+
+Yes. Sometimes there are classes, files, or entire namespaces that we don't want to expose to customers via the public-facing docs for various reasons. Typically, if customers don't need to know about something, we don't want it cluttering up the docs. Even if we don't add Doxygen-style comments to an item, Doxygen still scans the code and lists the item in the docs.
+
+Our Doxygen config **already excludes** the `AZ::Internal` namespace. Code marked as private, files with "test" in the name, files with "ReleaseNotes" in the name, and `docs.h` files. You don't need to do anything to get Doxygen to ignore those items, because they're already taken care of. In all other cases, follow the instructions in [Excluding code from the documentation](#excluding-code-from-the-documentation).
```

Comment:
We should remove this section mentioning a doxygen config.
There is no Doxygen config currently and if one comes on line in the future, this section can be updated then.

We can just mention that any Code within an "Internal" namespace is private in addition to the scenarios mentioned.

While O3DE uses DoxyGen commenting, Doxygen isn't actually run on over it at this time

Reply:
That's a bummer, I thought we could still generate it 😞 Ah well, I'll kill this for now

Reply:
There's actually quite a few more references to Doxygen throughout the doc, I take it I should remove all of them for now?

Reply:
Hmm I thinking I add this version as-is, and we do a second pass removing all the doxygen content so we have a log of it if we want to get it back